crates/cubecl-linalg/src/matmul/kernels/matmul/algorithm/selector.rs

```diff
@@ -0,0 +1,286 @@
+                    StandardAlgorithm<EG, ES, EA, S2x2x2, Accelerated16x16x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                4 => matmul_cube_preparation::<
+                    R,
+                    EG,
+                    StandardAlgorithm<EG, ES, EA, S4x4x2, Accelerated16x16x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                8 => matmul_cube_preparation::<
+                    R,
+                    EG,
+                    StandardAlgorithm<EG, ES, EA, S8x8x2, Accelerated16x16x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                _ => panic!("No configuration found for this stage size. "),
+            },
+            (32, 8) => match stage_size_m_n {
+                1 => matmul_cube_preparation::<
+                    R,
+                    EG,
+                    StandardAlgorithm<EG, ES, EA, S1x1x2, Accelerated32x8x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                2 => matmul_cube_preparation::<
+                    R,
+                    EG,
+                    StandardAlgorithm<EG, ES, EA, S2x2x2, Accelerated32x8x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                4 => matmul_cube_preparation::<
+                    R,
+                    EG,
+                    StandardAlgorithm<EG, ES, EA, S4x4x2, Accelerated32x8x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                8 => matmul_cube_preparation::<
+                    R,
+                    EG,
+                    StandardAlgorithm<EG, ES, EA, S8x8x2, Accelerated32x8x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                _ => panic!("No configuration found for this stage size. "),
+            },
+            (8, 32) => match stage_size_m_n {
+                1 => matmul_cube_preparation::<
+                    R,
+                    EG,
+                    StandardAlgorithm<EG, ES, EA, S1x1x2, Accelerated8x32x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                2 => matmul_cube_preparation::<
+                    R,
+                    EG,
+                    StandardAlgorithm<EG, ES, EA, S2x2x2, Accelerated8x32x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                4 => matmul_cube_preparation::<
+                    R,
+                    EG,
+                    StandardAlgorithm<EG, ES, EA, S4x4x2, Accelerated8x32x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                8 => matmul_cube_preparation::<
+                    R,
+                    EG,
+                    StandardAlgorithm<EG, ES, EA, S8x8x2, Accelerated8x32x16<ES, EA>>,
+                >(client, lhs, rhs, out, problem),
+                _ => panic!("No configuration found for this stage size. "),
+            },
+            _ => panic!("No configuration found for instruction shapes."),
+        }
+    }
+}
```

Comment:
Maybe this can be macros, it's pattern matching so that might be cleaner in a way, very optional.